### ISSUE TITLE
Files copied in unbuffered mode are not closed

### ISSUE THREAD

**Issue by agross** [gulpjs/vinyl-fs#56]
Related issue: https://github.com/gulpjs/gulp/issues/910

I'm trying to copy several thousands of images with streaming (to save memory) using vinyl-fs. The code is relatively simple:

```js
var vfs = require('vinyl-fs');
vfs
  .src('source/data/images/*.jpg', { buffer: false })
  .pipe(vfs.dest('dist/vfs'));
```

The process stops after copying 1985 files (on Windows, less on Fedora 21) of 3000 files. Using Process Explorer (Windows, ~`lsof`) and  `NODE_DEBUG=GFS` I found that node never closes the files written to `vfs.dest`.

How can I enforce the file descriptors to be freed after my last `pipe()`?

**Comment by contra** [comment]
Hmm really interesting. I will take a look at this

**Comment by rickihastings** [comment]
Any update on this? Also experiencing an issue where the process stops at round 70MB of usage, it seems to be running out of FD's (windows).

**Comment by phated** [comment]
I believe this is fixed by #121 - I think we were trying to open too many file descriptors up front and that killed the process because node seems to handle closing of the stream just fine.
